public/electron.js

```diff
@@ -171,42 +166,63 @@ ipc.on('trigger', (event, args) => {
 
 // INCREMENTAL FILE SAVING
 let stream = false
-let fileName = ''
-let filePath = ''
-let directoryPath = ''
+let fileCreated = false
+let preSavePath = ''
 let participantID = ''
 let studyID = ''
+let today = new Date()
 let startTrial = -1
 
+let savePath = ""
+
+/**
+ * Abstracts constructing the filepath for saving data for this participant and study.
+ * @returns {string} The filepath.
+ */
+const getSavePath = (participantID, studyID) => {
+  const desktop = app.getPath('desktop')
+  const name = app.getName()
+  const today = new Date()
+  const date = today.toISOString().slice(0, 10)
+  return path.join(
+    desktop,
+    studyID,
+    participantID,
+    date,
+    name
+  )
+}
+
+const getFullPath = (fileName) => {
```

Comment:
with javascript scoping rules, if `savePath` changes, is this function correct - or is the `savePath` at the time of the function creation captured in the function?